[ADD] Productivity: IoT - Windows IoT doc

tiku-odoo opened the pull request:
Doc Task: https://www.odoo.com/web#id=3057207&menu_id=4720&cids=3&action=333&active_id=3835&model=project.task&view_type=form

@odoo/us-doc-review  This doc is ready for your review when you have a moment. 

Thanks in advance for your help on this.

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/4025).

StraubCreative commented:
Quick item on the CI check @tiku-odoo https://runbot.odoo.com/runbot/build/27793011

tiku-odoo commented:
@StraubCreative I corrected this build error. Thanks for catching.

hojo-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,121 @@
+Uninstalling Windows IoT
+------------------------
+
+Uninstalling the Windows virtual IoT is done through the Windows program manager. Search in any
```
Add "box"
```suggestion
Uninstalling the Windows virtual IoT box is done through the Windows program manager. Search in any
```

hojo-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,121 @@
+Uninstalling Windows IoT
+------------------------
+
+Uninstalling the Windows virtual IoT is done through the Windows program manager. Search in any
+Windows version for ''program''. Select :guilabel:`Add or Remove Programs` located in the control
+panel. Search for ``Odoo`` and click the :guilabel:`three dot menu` to uninstall.
```
Rephrase; also, just searching "Odoo" may bring up multiple programs, so you should specify the exact name of the program being uninstalled
```suggestion
Uninstalling the Windows virtual IoT box is done through the Windows program manager. From the
:guilabel:`Start` menu, search for ''program''. Select :guilabel:`Add or Remove Programs` . In the window that appears, search for ``Odoo``, click the :guilabel:`three dot menu`, then select :guilabel:`Uninstall`.
```

tiku-odoo replied:
Odoo is the name of the program that should be uninstalled. I tested this a couple times. What did you encounter?

tiku-odoo replied:
Did you encounter multiple Odoo programs? The uninstall should be from Odoo and there are other programs that will be uninstalled.

hojo-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,121 @@
+Windows version for ''program''. Select :guilabel:`Add or Remove Programs` located in the control
+panel. Search for ``Odoo`` and click the :guilabel:`three dot menu` to uninstall.
+
+Confirm the uninstallation and follow the steps to uninstall through the Odoo 16.0 Uninstall guide.
```
Decaptialize uninstall; also, link to the uninstall guide
```suggestion
Confirm the uninstallation and follow the steps to uninstall through the Odoo 16.0 uninstall guide.
```

hojo-odoo reviewed (changes requested):
Hey Tim,

Good job on this. 

Thanks,
hojo

tiku-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,121 @@
+Windows virtual IoT box
+=======================
+
+Use Windows IoT software to connect devices to an Odoo database. The Virtual IoT box is a computer
```
Thanks, I've added this in.

tiku-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,121 @@
+#. Ensure that the :guilabel:`Start Odoo` box is checked and click :guilabel:`Finish`.
+
+#. After installation, the Odoo server will run and automatically open ``localhost:8069`` on your
+   web browser. The webpage should load into the IoT Box Configuration Page.
```
I've reworded this, thanks for catching this.

tiku-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,121 @@
+Virtual Windows IoT box not showing on the database
+---------------------------------------------------
+
+In some instances a manual restart of the physical :abbr:`IoT (Internet of Things)` box can resolve
+the issue. For the Windows virtual :abbr:`IoT (Internet of Things)` box a manual restart of the Odoo
+server can resolve database connection issues.
+
+To restart the virtual Windows IoT server:
+
+#. Type "Services" into the :guilabel:`Search Bar`
+#. Select the :menuselection:`Services` App and scroll down to the :guilabel:`Odoo` Service.
+#. Right click on :guilabel:`Odoo` and select :guilabel:`Start`. This action will manually restart
+   the Odoo IoT server.
```
I've deleted it out, thanks.

tiku-odoo commented:
@lse-odoo Can you review this doc when you have a moment on the Windows Virtual IoT?

Thanks :+1:

lse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,112 @@
+- Odoo 16 database
+- :abbr:`IoT (Internet of Things)` compatible devices
+- Device drivers for Windows.
+- Windows computer (laptop, desktop, server)
```
might be worth to precise which OS version we support (Windows 10 and > ?). I already foreseen some Windows XP customers complaining that it does not works (or 32bits/64 bits issues). Might be worth to check which Windows versions are really compatible

tiku-odoo replied:
Noted, I added that Odoo recommends an updated version of Windows as older versions can cause issues. This is a good point, thanks for adding this note.

lse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,112 @@
+==============
+
+- Odoo 16 database
+- :abbr:`IoT (Internet of Things)` compatible devices
```
Might be worth to add a .. seealso to the section with iot devices (or to the compatibility table I mentioned earlier)

tiku-odoo replied:
I've added the following: 

`- Odoo 16 database or any version above
- :abbr:`IoT (Internet of Things)` compatible devices

  ..seealso::
    `Odoo's compatible IoT devices <https://www.odoo.com/app/iot-hardware>`_`

lse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,112 @@
+      Odoo's Windows virtual IoT software shouldn't be installed inside any of the Window's User's
+      directories. Doing so won't allow for Nginx to initialize.
+
+   Choosing ``C:\odoo`` as the install location will allow for the Nginx server to start. Click
```
don't know if it will allow you if the "odoo" folder does not exists in C:. Can you confirm to me that it will create the folder if it does not exists ?

lse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,112 @@
+   web browser. The webpage should display the :abbr:`IoT (Internet of Things)` Box configuration
+   page.
+
+#. Connect devices to Windows computer.
```
```suggestion
#. Connect IoT devices to the Windows computer.
```

lse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,112 @@
+Virtual Windows IoT box not showing on the database
+---------------------------------------------------
+
+In some instances a manual restart of the physical :abbr:`IoT (Internet of Things)` box can resolve
+the issue. For the Windows virtual :abbr:`IoT (Internet of Things)` box a manual restart of the Odoo
+server can resolve database connection issues.
+
+To restart the virtual Windows IoT server:
+
+#. Type "Services" into the :guilabel:`Search Bar`
+#. Select the :menuselection:`Services` App and scroll down to the :guilabel:`Odoo` Service.
+#. Right click on :guilabel:`Odoo` and select :guilabel:`Start` or :guilabel:`Restart`. This action
+   will manually restart the Odoo IoT server.
```
As other issues might need to restart the IoT box, maybe it would be worth to have its dedicated section `Restart Windows IoT` and a link here ?

tiku-odoo replied:
I've added an anchor in here and also linked it on line 73 as a possible route to follow should the local host not populate.

lse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,112 @@
+#. Right click on :guilabel:`Odoo` and select :guilabel:`Start` or :guilabel:`Restart`. This action
+   will manually restart the Odoo IoT server.
+
+Windows Firewall
```
In practice, the "Windows firewall" is not the only cause, most of the anti-virus software will have their own firewall system. As such, maybe we should be more generic

tiku-odoo replied:
Noted, I've edited this to be more generic, good catch @lse-odoo

lse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,112 @@
+Windows Firewall
+----------------
+
+The Windows virtual :abbr:`IoT (Internet of Things)` box software may not connect to the :abbr:`LAN
+(Local Area Network)` due to the Windows Firewall preventing the connection. Consult your local IT
```
Technically incorrect, the firewall can not prevent a device to connect in a network, it will blocked incoming and/or outgoing requests made with/from the device
```suggestion
The Windows virtual :abbr:`IoT (Internet of Things)` box software may be reachable in the :abbr:`LAN
(Local Area Network)` due to the Windows Firewall preventing the connection. Consult your local IT
```

lse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,112 @@
+The Windows virtual :abbr:`IoT (Internet of Things)` box software may not connect to the :abbr:`LAN
+(Local Area Network)` due to the Windows Firewall preventing the connection. Consult your local IT
+support to make exceptions in the :abbr:`OS (Operating System)`.
+
```
Could you check if "Network discovery" must be enabled in order to make it work on windows ?
See: https://www.makeuseof.com/windows-network-discovery-turn-on-off/
It would need to be check on both the Windows IoT computer and a Windows computer trying to reach the Windows IoT computer.
If it blocks connections, it might be worth to add a section asking to check to activate it.

tiku-odoo replied:
This gets too much into network configuration, I think this is enough info. Thanks.

tiku-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,112 @@
+computer. This requires a Windows operating system with an Odoo 16 or later database.
+
+The Windows Virtual :abbr:`IoT (Internet of Things)` box works the same way as a physical
+:abbr:`IoT (Internet of Things)` box, with the ability to run most of the same devices. All
+:abbr:`POS (Point of Sale)` devices work with it, such as a scale or printer. Payment terminals will
+also work, but it should be noted that :abbr:`MRP (Material Requirement Planning)` devices are not
+compatible. These include cameras or measurement tools.
```
This is a good idea, except I don't have the ability to do any testing (lack of devices), this information would need to come from the experts in SF and BE. Is that something that you can do? CC: @dara-odoo 

Thanks, 
Tim

dara-odoo replied:
Hey @tiku-odoo ! I can perform some testing to see which devices are compatible with the virtual IoT Software! The only things I cannot test are the following: 

1. Weighing scale (our scale in SF doesn't work, even when connected to the traditional IoT Box.
2. Any MRP tools like measurement tools; we don't have that here. 

Someone from BE should be able to test the rest (assuming there is at least one Windows machine over there).  I'll let you know what's compatible through here in a few days!

dara-odoo replied:
Hey @tiku-odoo !

Apologies for the delay. I did some testing and I can confirm that the following devices are compatible with both Physical and Virtual IoT based on the devices we have in the SF office. For devices not tested, we would need BE's input. (also I am not too sure how to submit this in the devices.rst file): 

-----------
SCREEN

- HDMI device: 
- Physical IoT: Compatible.
- Virtual IoT: Not Working/Not Compatible. Device connected via HDMI is not being detected in the Virtual IoT Box's configuration page. 

- Integrated display (Laptop):
- Physical IoT: Compatible
- Virtual IoT: Not Working/Not Compatible. See above reason. 

-----------
MEASUREMENT TOOL

Not tested in SF office; Does not have required device to test this on Physical or Virtual IoT. 

-----------
CAMERA

- Logitech (as suggested on our IoT Devices list): Not tested in the SF office; Does not have required device to test this on Physical or Virtual IoT.
- Non-Logitech:
- Physical IoT: Compatible via USB only.
- Virtual IoT: Not Working/Not Compatible. Camera connected via USB is not being detected in the Virtual IoT Box's configuration page. 

-----------
FOOTSWITCH

Not tested in the SF office; Does not have required device to test this on Physical or Virtual IoT.

-----------
PRINTERS

Receipt Printers

- Epson Printers (with ESC/POS protocol): 
- Physical IoT: Compatible; either connected via USB or via Network
- Virtual IoT: Compatible; either connected via USB or via Network. You would need to install the required drivers for the Epson printer to work on Virtual IoT.  

- Star-compatible Printers (with STAR protocol): Not tested in the SF office; Does not have required device to test this on Physical or Virtual IoT.


Receipt Printers > Cash Registers

- Epson Compatible: Can only be connected to an Epson printer using the serial/network cable. See "Epson Printers" for compatibility.
- USB: Not Compatible. 

 
Desktop Printers 

- Brother Inkjet/Laserjet: 
- Physical IoT: Compatible; either connected via USB or via Network. Must be detectible from the integrated CUPS server on the Physical IoT.  
- Virtual IoT: Compatible; either connected via USB or via Network. You would need to install the required drivers for the Desktop printer to work on Virtual IoT (no integrated CUPS server to be auto-detected). 

Label Printers

- Zebra Printers (with ZPL protocol): 
- Physical IoT: Compatible via USB only. 
- Virtual IoT: Not Working/Not Compatible. Windows drivers allow the printer to be seen on the computer and on the virtual IoT Software, but is not able to utilized when a report is assigned to the printer. Receives an "IoT Box is disconnected" error. 

-----------
SCALE

Not tested in the SF office; Current scale at SF office does not connect to IoT Box properly, even if it's a suggested scale. Cannot test on Physical or Virtual IoT.

tiku-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,112 @@
+
+- Odoo 16 database
+- :abbr:`IoT (Internet of Things)` compatible devices
+- Device drivers for Windows.
```
Good point I've added in the link: https://www.odoo.com/app/iot-hardware

tiku-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,112 @@
+===============================================================
+Use Windows IoT software to connect devices to an Odoo database
```
I originally had something like you mentioned and the final editor recommended I change it.

tiku-odoo commented:
@samueljlieber This doc is ready for your technical review when you have a moment. Thanks for your review on this. 

Tim

lse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,134 @@
+#. Right click on :guilabel:`Odoo` and select :guilabel:`Start` or :guilabel:`Restart`. This action
+   will manually restart the Odoo IoT server.
+
+Firewalls
+---------
+
+The Windows virtual :abbr:`IoT (Internet of Things)` box software may not be reachable to the
+:abbr:`LAN (Local Area Network)` due to a firewall preventing the connection. Consult your local IT
+support to make exceptions (network discovery) in the :abbr:`OS (Operating System)` or firewall
+program. Windows has their own firewall as well as other virus protection do too.
```
From a support point of view, I am a bit afraid that this part will be overlooked by customers as it is "too technical".
Don't you think it would be better to use an actual example ? Like:

"I can see the IoT box homepage from the Windows virtual computer, but can not access it from other computers/mobiles/tablet in the same network"

speaking about LAN will confuse most of the customers and we (the support) will likely lose time redirecting then to this part of the documentation. It might also be a good occasion to add a .seealso to
https://www.odoo.com/documentation/16.0/applications/productivity/iot/config/troubleshooting.html#my-iot-box-is-connected-to-my-database-but-cannot-be-reached

What do you think ?

tiku-odoo replied:
I've added this example into the doc. Thanks for suggesting @lse-odoo

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,139 @@
+Connecting the Windows IoT software to an Odoo database
+=======================================================
+
+A Virtual IoT box is a computer program that needs to be downloaded and installed on a Windows
```
```suggestion
A *Virtual IoT box* is a computer program that needs to be downloaded and installed on a Windows
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,139 @@
+#. Navigate to the Odoo 16 or higher installation package for Enterprise/Community Windows at
+   `Odoo's Download Page <https://odoo.com/download>`_.
+
+#. Install and setup the Odoo .exe file.
```
```suggestion
#. Install and setup the Odoo :file:`.exe` file.
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,139 @@
+
+#. Install and setup the Odoo .exe file.
+
+   After the instructions screen click :guilabel:`Next` to start the installation. Agree to the
```
```suggestion
   After the :guilabel:`Instructions` screen click :guilabel:`Next` to start the installation. Agree
   to the :abbr:`TOS (Terms of Service)`.
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,139 @@
+
+   For reference, the following should be installed:
+
+   - **Odoo server**
+   - **Odoo IoT**
+   - **Nginx WebServer**
+   - **Ghostscript interpreter**
```
I had to add another space for these sub-bullet points to get the sentence after (in the parent bullet) to format correctly.

```suggestion
    - **Odoo server**
    - **Odoo IoT**
    - **Nginx WebServer**
    - **Ghostscript interpreter**
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,139 @@
+   - **Odoo IoT**
+   - **Nginx WebServer**
+   - **Ghostscript interpreter**
+
```
Can remove this extra line
```suggestion
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,139 @@
+      Odoo's Windows virtual IoT software shouldn't be installed inside any of the Window's User's
+      directories. Doing so won't allow for Nginx to initialize.
+
+   Choosing ``C:\odoo`` as the install location will allow for the Nginx server to start. Click
```
Only need one ` for a literal

```suggestion
   Choosing `C:\odoo` as the install location will allow for the Nginx server to start. Click
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,139 @@
+
+#. Ensure that the :guilabel:`Start Odoo` box is checked and click :guilabel:`Finish`.
+
+#. After installation, the Odoo server will run and automatically open ``localhost:8069`` on your
```
```suggestion
#. After installation, the Odoo server will run and automatically open `localhost:8069` on your
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,139 @@
+#. Windows should automatically detect the device because the driver is pre-installed on
+   the computer. If not, search for and install the Windows driver for the device.
+
+#. Refresh the :abbr:`IoT (Internet of Things)`  Box Configuration Page and verify the device is
+   seen on the configuration page. If not, reload the handlers through the configuration page.
```
```suggestion
#. Refresh the :abbr:`IoT (Internet of Things)` Box :guilabel:`Configuration` page and verify the
   device is seen on the :guilabel:`Configuration` page. If not, reload the handlers through the
   :guilabel:`Configuration` page.
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,139 @@
+   (manually using the Token).
+
+   .. seealso::
+      :doc:`connect`
```
```suggestion
      :doc:`/applications/productivity/iot/config/connect`
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,139 @@
+
+#. Use the devices connected to :abbr:`IoT (Internet of Things)` as you normally would.
+
+
```
```suggestion
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,139 @@
+Troubleshooting
+===============
+
+.. _restart_windows_iot:
```
Let's add the app in the anchor per [the guidelines](https://www.odoo.com/documentation/15.0/contributing/documentation/rst_guidelines.html#prefix-hyperlink-targets-with-application-names)

```suggestion
.. _productivity/restart_windows_iot:
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,139 @@
+
+To restart the virtual Windows IoT server:
+
+#. Type "Services" into the :guilabel:`Search Bar`
```
```suggestion
#. Type `Services` into the :guilabel:`Search Bar`
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,139 @@
+------------------------
+
+Uninstalling the Windows virtual :abbr:`IoT (Internet of Things)` box is done through the Windows
+program manager. Search in any Windows version for ''program''. Select :guilabel:`Add or Remove
+Programs` located in the control panel. Search for ``Odoo`` and click the :guilabel:`three dot menu`
+to uninstall.
```
```suggestion
program manager. Search in any Windows version for `program`. Select :menuselection:`Add or Remove
Programs` located in the :guilabel:`Control Panel`. Search for `Odoo` and click the :guilabel:`⋮
(three dot menu)` to uninstall.
```

samueljlieber reviewed (changes requested):
Hi @tiku-odoo 👋 I've made some technical changes on `windows_iot.rst`, mostly very small formatting changes. Please review the changes below and let me know if you approve and I will tag this PR for final review. I will push these changes up in a commit after this review.

samueljlieber commented:
Implemented [technical changes](https://github.com/odoo/documentation/pull/4025#pullrequestreview-1406355174) and corrected commit title in https://github.com/odoo/documentation/commit/bfb1de1e0adfd1cf88b466b0f51c6d52f82f98b9

tiku-odoo commented:
@samueljlieber The changes you made are good to push to Final review. Thanks for your technical review. 

Tim :+1:

samueljlieber commented:
Hi @StraubCreative this PR is ready for your final review 🙂

StraubCreative reviewed (changes requested):
@tiku-odoo please fix the 100th character line break issues here and reconsider your lists approach to writing thorough documentation.

Lists are ideally used for [form walkthroughs](https://www.odoo.com/documentation/16.0/applications/inventory_and_mrp/manufacturing/quality_control/quality_control_points.html) or for sets of [quick, successive instructions](https://www.odoo.com/documentation/16.0/applications/inventory_and_mrp/inventory/management/products/strategies.html#id2) that _must_ be done in order; lists do not suffice as the main body content.

Tag me again when it's ready for another look, thanks!

cc: @hojo-odoo

tiku-odoo commented:
@StraubCreative 

Thanks for your preliminary review. It's ready for another look! 

CC: @hojo-odoo

tiku-odoo commented:
@StraubCreative I've removed the lists and made the document flow much better, added some headings and example blocks, similar to the Sendcloud, OAuth docs. This is ready for your review again. 

Thanks, Tim :+1:

tiku-odoo commented:
@samueljlieber Can you review this doc when you have a moment? Thanks, Tim

brse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,178 @@
+`Firewall`. Then open the :menuselection:`Windows Defender Firewall` program. In the left menu,
+navigate to :guilabel:`Advanced Settings`.
+
+Following selecting :guilabel:`Advanced Settings`, select :guilabel:`Inbound Rules` in the left
+menu. Now in the right menu column, under :guilabel:`Inbound Rules`, and click on :guilabel:`New
+Rule`.
```
```suggestion
Once :guilabel:`Advanced Settings` have been selected, click :guilabel:`Inbound Rules` in the left-hand
menu. Then, in the right-hand menu column (under :guilabel:`Inbound Rules`), click on :guilabel:`New
Rule` to create a new rule.
```

brse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,178 @@
+Configure new rule
+******************
+
+On the :menuselection:`Rule Type` screen, select :guilabel:`Port`. Then click :guilabel:`Next`.
```
```suggestion
From the *Rule Type* screen, select the :guilabel:`Port` option, and click :guilabel:`Next`.
```
I *think* for forms/pages/screens we're on, we wouldn't use `:menuselection:` tags. If I'm wrong though ignore that part of the change!

tiku-odoo replied:
I'm going to leave this in place, as it's a wizard and can be seen approved by ZST in Line 33 of [google_oauth.rst](https://github.com/odoo/documentation/pull/3965)

brse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,178 @@
+On the :menuselection:`Rule Type` screen, select :guilabel:`Port`. Then click :guilabel:`Next`.
+Leave the rule application to :guilabel:`TCP` on the :menuselection:`Protocol and Ports` page.
+Select :guilabel:`Specific Local Ports` for :guilabel:`ports`. In the text box, type in `8069`.
+Finally click :guilabel:`Next` to continue to the next step.
```
```suggestion
Finally, click :guilabel:`Next` to continue to the next step.
```

brse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,178 @@
+******************
+
+On the :menuselection:`Rule Type` screen, select :guilabel:`Port`. Then click :guilabel:`Next`.
+Leave the rule application to :guilabel:`TCP` on the :menuselection:`Protocol and Ports` page.
+Select :guilabel:`Specific Local Ports` for :guilabel:`ports`. In the text box, type in `8069`.
```
```suggestion
From the *Protocol and Ports* page, leave the rule application as :guilabel:`TCP`. Then,
select :guilabel:`Specific Local Ports` for the :guilabel:`ports` option. In the text box, type in `8069`.
```
Again, if my suggestion above for referencing forms/screens/pages is wrong, disregard that part of this change

brse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,178 @@
+
+On the :menuselection:`Actions` page, select :guilabel:`Allow the connection` and click
+:guilabel:`Next`. The following page on the :menuselection:`Rule Configuration` wizard is the
+:menuselection:`Profile` page. On this page select whichever connection type applies to the network
```
```suggestion
:menuselection:`Profile` page. On this page, select whichever connection type applies to the network
```

brse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,178 @@
+On the :menuselection:`Actions` page, select :guilabel:`Allow the connection` and click
+:guilabel:`Next`. The following page on the :menuselection:`Rule Configuration` wizard is the
+:menuselection:`Profile` page. On this page select whichever connection type applies to the network
+the Windows machine is operating on. Ideally, select :guilabel:`Private` only
+connections. Click :guilabel:`Next` to continue.
```
Why only private connections? A quick sentence to explain could help

brse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,178 @@
+connections. Click :guilabel:`Next` to continue.
+
+Finally, name the rule. For example this can be `Odoo`. Optionally, add a brief description in the
+:guilabel:`Description` field. Click :guilabel:`Finish` to complete the *rule configuration*. Now
```
```suggestion
:guilabel:`Description` field. Click :guilabel:`Finish` to complete the *rule configuration*. Now,
```

brse-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,178 @@
+------------------------
+
+Uninstalling the Windows virtual :abbr:`IoT (Internet of Things)` box is done through the Windows
+program manager. Search in any Windows version for ''program''. Select :guilabel:`Add or Remove
```
```suggestion
program manager. Using any Windows version, search for for ''program''.  Then, select :guilabel:`Add or Remove
```

brse-odoo reviewed (commented):
Hey @tiku-odoo, I've just finished my review of the new section. Great job! All my suggestions were pretty minor grammatical stuff, but I think a sentence or two at the start of each section to better explain concepts would be helpful.

Another thing that I didn't comment but might be worth considering -- can the H5 headings you use be H4s, instead? Only if it makes sense (I think H5 is really small). Otherwise it's not a big deal. Let me know if you have any questions!

tiku-odoo commented:
@samueljlieber This doc is ready for your technical review.

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+Pre-requisites
+==============
+
+The following items will be needed to complete the Windows IoT installation.
```
Keep consistent abbreviation use throughout doc :)
```suggestion
The following items will be needed to complete the Windows :abbr:`IoT (Internet of Things)`
installation.
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+Connecting the Windows IoT software to an Odoo database
+=======================================================
+
+A Virtual IoT box is a computer program that needs to be downloaded and installed on a Windows
```
Unless it is a proper noun or is the name of the tool/module we shouldn't capitalize *virtual* in virtual IoT box. Looks like there are only a few of these throughout the doc that have the `V` capitalized :)
```suggestion
A virtual IoT box is a computer program that needs to be downloaded and installed on a Windows
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+---------------------------------
+
+To begin the installation, navigate to the Odoo 16 or higher installation package for Enterprise or
+Community Windows at `Odoo's Download Page <https://odoo.com/download>`_. Next, Install and setup
```
maintain sentence case in the hyperlink, and lowercase _install_
```suggestion
Community Windows at `Odoo's download page <https://odoo.com/download>`_. Next, install and setup
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+
+To begin the installation, navigate to the Odoo 16 or higher installation package for Enterprise or
+Community Windows at `Odoo's Download Page <https://odoo.com/download>`_. Next, Install and setup
+the Odoo `.exe` file. After the instructions screen, click :guilabel:`Next` to start the
```
Totally optional - Im a fan of using the :file: tag for file extentions and filenames :)
```suggestion
the Odoo :file:`.exe` file. After the instructions screen, click :guilabel:`Next` to start the
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+
+Ensure that the :guilabel:`Start Odoo` box is checked and click :guilabel:`Finish`. After
+installation, the Odoo server will run and automatically open `http://localhost:8069` on your web
+browser. The webpage should display the :abbr:`IoT (Internet of Things)` *box home page*
```
Missing period and can remove italic formatting
```suggestion
browser. The webpage should display the :abbr:`IoT (Internet of Things)` box home page.
```

tiku-odoo replied:
@StraubCreative This is the label of the page, I originally didn't have this italicized, but the expert pointed this out. I'm okay with either way, not a big deal in my opinion. We will have to implement in the IoT bundle PR if there are changes though (PR  #3673). I can double check the label as well on this.

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+(manually using the Token).
+
+.. seealso::
+   :doc:`connect`
```
please use global relative paths for doc tags :)
```suggestion
   :doc:`/applications/productivity/iot/config/connect`
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+Troubleshooting
+===============
+
+.. _restart_windows_iot:
```
please add the app name to the custom anchor (don't forget to update the corresponding :ref: tag)
```suggestion
.. _iot/restart_windows_iot:
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+
+.. _restart_windows_iot:
+
+Restart Windows IOT box
```
```suggestion
Restart Windows IoT box
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+automatically detect the device because the driver is pre-installed on the computer. If not, search
+for and install the Windows driver for the device.
+
+Following connecting devices to the computer, refresh the :abbr:`IoT (Internet of Things)` *box
+home page* and verify the device is seen. If not, reload the handlers through the :abbr:`IoT
+(Internet of Things)` *box home page*.
```
early 100th char line break
```suggestion
Following connecting devices to the computer, refresh the :abbr:`IoT (Internet of Things)` *box home
page* and verify the device is seen. If not, reload the handlers through the :abbr:`IoT (Internet of
Things)` *box home page*.
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+
+To restart the virtual Windows IoT server:
+
+#. Type "Services" into the :guilabel:`Search Bar`
```
Please use pre formatting for user inputs. Also missing a period

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+To restart the virtual Windows IoT server:
+
+#. Type "Services" into the :guilabel:`Search Bar`
+#. Select the :menuselection:`Services` App and scroll down to the :guilabel:`Odoo` Service.
```
```suggestion
#. Select the :menuselection:`Services` App and scroll down to the :guilabel:`Odoo` service.
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+Making an exception on Windows Defender
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It's possible to allow other devices to access the Windows Virtual :abbr:`IoT (Internet of Things)`
```
```suggestion
It's possible to allow other devices to access the Windows virtual :abbr:`IoT (Internet of Things)`
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+******************
+
+On the :menuselection:`Rule Type` screen, select :guilabel:`Port`. Then click :guilabel:`Next`.
+From the :menuselection:`Protocol and Ports` page leave the rule application to :guilabel:`TCP`.
```
Early 100th char line break

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+On the :menuselection:`Actions` page, select :guilabel:`Allow the connection` and click
+:guilabel:`Next`. The following page on the :menuselection:`Rule Configuration` wizard is the
+:menuselection:`Profile` page. On this page, select whichever connection type applies to the network
+the Windows machine is operating on. Ideally, select :guilabel:`Private` only
+connections. The *Private* connection type is the most secure connection while allowing the selected
```
early 100th char line breaks

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+
+Finally, assign a new, unique name to the rule. For example, this name can be `Odoo`. Optionally,
+add a brief description in the :guilabel:`Description` field. Click :guilabel:`Finish` to complete
+the *rule configuration*. Now, the new rule is active and devices can connect to the Windows Virtual
```
```suggestion
the *rule configuration*. Now, the new rule is active and devices can connect to the Windows virtual
```

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,180 @@
+------------------------
+
+Uninstalling the Windows virtual :abbr:`IoT (Internet of Things)` box is done through the Windows
+program manager. Using any Windows version, search for for `program`.  Then, select :guilabel:`Add
```
duplicate word
```suggestion
program manager. Using any Windows version, search for `program`.  Then, select :guilabel:`Add
```

samueljlieber reviewed (changes requested):
Hi @tiku-odoo I have a handful of technical changes for this PR, please see them below and then tag me again for another review once these edits have been made. Thank you! 🙂

Let me know if you have any questions!

tiku-odoo commented:
@samueljlieber This doc is ready for another review. Thanks for you help on this doc! :+1: 

Tim

samueljlieber commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,181 @@
+
+Ensure that the :guilabel:`Start Odoo` box is checked and click :guilabel:`Finish`. After
+installation, the Odoo server will run and automatically open `http://localhost:8069` on your web
+browser. The webpage should display the :abbr:`IoT (Internet of Things)` *box home page*
```
```suggestion
browser. The webpage should display the :abbr:`IoT (Internet of Things)` *box home page*.
```

samueljlieber replied:
Still missing a period here 🙂

samueljlieber reviewed (approved):
Hi @tiku-odoo this PR is looking really great - nice job! I just found one missing period that remained. Please make this correction before tagging for final review, other than that I feel this PR is ready for final review. Thanks Tim! 🙂

tiku-odoo commented:
@StraubCreative  This doc is ready for your review. Thanks! :+1:

tiku-odoo commented:
@StraubCreative When you have a moment can you review this doc? It is ready for final review! 

Thanks.

StraubCreative reviewed (approved):
Hi @tiku-odoo 

Pretty fun and straight-forward doc. I feel like I could figure this out easily with what you have here 🙂 

I have a few suggestions below, however, these are all for your consideration and _are entirely optional_. It might be better to merge right away and improve later since this is a brand new doc...

With whatever you choose (making the suggested changes in another commit or just merging right away), just issue the merge command ( `[at]robodoo r+` ) whenever you're ready 🚀

tiku-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,181 @@
+- Device drivers for Windows.
+
+  .. note::
+     Odoo recommends using an updated, recent version of Windows as some older operating systems can
```
I've added in to use 10/11.

tiku-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,181 @@
+
+Next, connect the :abbr:`IoT (Internet of Things)` devices to the Windows computer. Windows should
+automatically detect the device because the driver is pre-installed on the computer. If not, search
+for and install the Windows driver for the device.
```
Not that I know of; these are Windows-based drivers. They can vary by the device manufacturer.

tiku-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,181 @@
+automatically detect the device because the driver is pre-installed on the computer. If not, search
+for and install the Windows driver for the device.
+
+Following connecting devices to the computer, refresh the :abbr:`IoT (Internet of Things)` *box home
+page* and verify the device is seen. If not, reload the handlers through the :abbr:`IoT (Internet of
+Things)` *box home page*.
```
Noted, I've rechecked the label and there is no title, nor label. I've removed the formatting per Sam @samueljlieber 
![iot-configuration](https://github.com/odoo/documentation/assets/91896312/3f82a21f-b88e-4030-8ed8-a4aa94e048c8)

tiku-odoo replied:
New formatting: :abbr:`IoT (Internet of Things)` box homepage

tiku-odoo commented on content/applications/productivity/iot/config/windows_iot.rst:
```diff
@@ -0,0 +1,181 @@
+
+To restart the virtual Windows IoT server:
+
+#. Type `Services` into the :guilabel:`Search Bar`.
```
Added in Windows

tiku-odoo commented:
@robodoo delegate+

robodoo commented:
I'm sorry, @tiku-odoo: you can't review+.
